app/components/HoverPreview.js

```diff
@@ -0,0 +1,182 @@
+// @flow
+import * as React from "react";
+import { inject } from "mobx-react";
+import { transparentize } from "polished";
+import Editor from "components/Editor";
+import styled from "styled-components";
+import { Portal } from "react-portal";
+import { fadeAndSlideIn } from "shared/styles/animations";
+import isInternalUrl from "utils/isInternalUrl";
+import { parseDocumentSlugFromUrl } from "shared/utils/parseDocumentIds";
+import DocumentsStore from "stores/DocumentsStore";
+import DocumentMeta from "components/DocumentMeta";
+
+const DELAY_OPEN = 500;
+const DELAY_CLOSE = 500;
+
+type Props = {
+  node: HTMLAnchorElement,
+  event: MouseEvent,
+  documents: DocumentsStore,
+  onClose: () => void,
+};
+
+function HoverPreview({ node, documents, onClose, event }: Props) {
+  const bounds = node.getBoundingClientRect();
+
+  // previews only work for internal doc links for now
+  if (!isInternalUrl(node.href)) {
+    return null;
+  }
+
+  const slug = parseDocumentSlugFromUrl(node.href);
+
+  // $FlowFixMe
```

Comment:
I hate all these, but we need to upgrade flow and that should happen in a separate PR.